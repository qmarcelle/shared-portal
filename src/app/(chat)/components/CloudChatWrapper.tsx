'use client';

// CloudChatWrapper injects the Genesys Cloud Messenger SDK and manages Messenger-specific events.
// All script loading, Messenger events, and errors are logged for traceability and debugging.

import { useEffect } from 'react';
import { useChatStore } from '../stores/chatStore';

export default function CloudChatWrapper() {
  const { userData } = useChatStore();

  useEffect(() => {
    // Build chatSettings from env and userData
    window.chatSettings = {
      bootstrapUrl: process.env.NEXT_PUBLIC_GENESYS_BOOTSTRAP_URL || '',
      widgetUrl: process.env.NEXT_PUBLIC_GENESYS_WIDGET_URL || '',
      clickToChatJs: process.env.NEXT_PUBLIC_GENESYS_CLICK_TO_CHAT_JS || '',
      clickToChatEndpoint: process.env.NEXT_PUBLIC_CLICK_TO_CHAT_ENDPOINT || '',
      chatTokenEndpoint: process.env.NEXT_PUBLIC_CHAT_TOKEN_ENDPOINT || '',
      coBrowseEndpoint: process.env.NEXT_PUBLIC_COBROWSE_LICENSE_ENDPOINT || '',
      opsPhone: process.env.NEXT_PUBLIC_OPS_PHONE || '',
      opsPhoneHours: process.env.NEXT_PUBLIC_OPS_HOURS || '',
      ...userData,
    };
    console.debug('[CloudChatWrapper] chatSettings:', window.chatSettings);
  }, [userData]);

  useEffect(() => {
    // Load Genesys Cloud bootstrap and widget scripts
    const bootstrapUrl = window.chatSettings?.bootstrapUrl || '';
    const widgetUrl = window.chatSettings?.widgetUrl || '';
    if (!bootstrapUrl) return;
    const bootstrapScript = document.createElement('script');
    bootstrapScript.src = bootstrapUrl;
    bootstrapScript.async = true;
    bootstrapScript.onload = () => {
      console.debug('[CloudChatWrapper] Genesys bootstrap loaded');
      if (!widgetUrl) return;
      const widgetScript = document.createElement('script');
      widgetScript.src = widgetUrl;
      widgetScript.async = true;
      widgetScript.onload = () => {
        console.debug('[CloudChatWrapper] Genesys widgets loaded');
      };
      widgetScript.onerror = (e) => {
        console.error('[CloudChatWrapper] Failed to load widgets script', e);
      };
      document.body.appendChild(widgetScript);
    };
    bootstrapScript.onerror = (e) => {
      console.error('[CloudChatWrapper] Failed to load bootstrap script', e);
    };
    document.body.appendChild(bootstrapScript);

    // Provide openGenesysChat logic
    window.openGenesysChat = () => {
      if (window.Genesys) {
        try {
          window.Genesys('command', 'Messenger.open');
        } catch (e) {
          console.error(
            '[CloudChatWrapper] Error opening Genesys Messenger',
            e,
          );
        }
      }
    };

    return () => {
      // Cleanup scripts and chatSettings
      document.body.removeChild(bootstrapScript);
      if (widgetUrl) {
        const widgetScript = document.querySelector(
          'script[src="' + widgetUrl + '"]',
        );
        if (widgetScript) document.body.removeChild(widgetScript);
      }
      if (window.chatSettings) delete window.chatSettings;
      if (window.openGenesysChat) delete window.openGenesysChat;
      console.debug('[CloudChatWrapper] Cleanup complete');
    };
  }, [userData]);

  return null;
}
