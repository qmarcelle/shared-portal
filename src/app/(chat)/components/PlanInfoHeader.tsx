'use client';
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { Title } from '@/components/foundation/Title';

/**
 * Show current plan when multiple plans exist
 */
export default function PlanInfoHeader() {
  const { formInputs } = useChatStore();
  if (formInputs.length <= 1) return null;
  const plan = formInputs.find((f) => f.id === 'PLAN_ID');

  return (
    <Title
      className="plan-info-header"
      text={`You are chatting about plan ${plan?.value || ''}`}
    />
  );
}
