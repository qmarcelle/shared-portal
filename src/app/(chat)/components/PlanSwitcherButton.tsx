'use client';
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { Button } from '@/components/foundation/Button';
import { usePlanStore } from '../../../userManagement/stores/planStore';

/**
 * "Switch Plan" inside pre-chat window
 */
export default function PlanSwitcherButton() {
  const { endChat } = useChatStore();
  const { plans, selectedPlanId, openPlanSwitcher } = usePlanStore();

  if (plans.length <= 1) return null;

  const handleSwitch = () => {
    endChat();
    openPlanSwitcher();
  };

  return (
    <Button
      label="Switch Plan"
      callback={handleSwitch}
      className="plan-switch-button"
      type="secondary"
    />
  );
}
