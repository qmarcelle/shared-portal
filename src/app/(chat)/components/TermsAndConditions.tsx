'use client';
// import TCs from '@/app/@chat/config/terms'; // { default: '…', lobA: '…', … }
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { RichText } from '@/components/foundation/RichText';

/**
 * Render T&C based on LOB
 */
export default function TermsAndConditions() {
  const { userData } = useChatStore();
  // const text = TCs[userData.LOB] || TCs.default;
  return (
    <RichText
      spans={['Terms and Conditions']}
      className="terms-and-conditions"
      type="body-2"
    />
  );
}
