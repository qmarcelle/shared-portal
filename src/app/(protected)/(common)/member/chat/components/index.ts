/**
 * Component Exports
 * Centralizes all component exports following the new consolidated structure
 */

// Main Components
export { ChatTrigger } from './ChatTrigger';
export { ChatWidget } from './ChatWidget';
export { PlanInfoHeader } from './PlanInfoHeader';

// Error Handling
export { ChatErrorBoundary } from './shared/ChatErrorBoundary';

// Integration Components (to be consolidated)
export { GenesysScripts } from './GenesysScripts';
