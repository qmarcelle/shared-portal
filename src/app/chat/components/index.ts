// Core Components
export * from './core/ChatButton';
export * from './core/ChatErrorBoundary';
export * from './core/ChatHeader';
export * from './core/ChatWidget';

// Business Hours Components
export * from './business-hours/BusinessHoursNotification';

// Eligibility Components
export * from './eligibility/EligibilityCheck';

// Plan Switcher Components will be added here when implemented
// export * from './plan-switcher/PlanSwitcher';
