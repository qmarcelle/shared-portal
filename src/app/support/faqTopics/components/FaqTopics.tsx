import { Column } from '@/components/foundation/Column';
import { Spacer } from '@/components/foundation/Spacer';
import { useEffect, useState } from 'react';
import { FaqHeaderCardDetails } from '../models/faq_header_card_details';
import { FaqTopicDetails } from '../models/faq_topic_details';
import { FaqTopicType } from '../models/faq_topic_type';
import { OtherFaqTopicDetails } from '../models/other_faq_topic_details';
import { SupportFaqTopicDetails } from '../models/support_faq_topic_details';
import { FaqCard } from './FaqCard';
import { FaqHeaderCard } from './FaqHeaderCard';
import { OtherFaqTopics } from './OtherFaqTopics';

let activeFaqTopic: { (faqType: string | null): void; (item: string): void };

export const FaqTopics = () => {
  const [topics, setTopics] = useState<FaqTopicDetails>();
  let faqType: string | null;
  useEffect(() => {
    const urlSearchString = window.location.search;
    const params = new URLSearchParams(urlSearchString);
    faqType = params.get('faqtype');
    activeFaqTopic(faqType);
  }, []);

  activeFaqTopic = (faqType: string | null) => {
    switch (faqType) {
      case 'Benefits':
      case 'Benefits & Coverage':
        console.log('topics  ----' + topics);
        return setTopics(
          SupportFaqTopicDetails.get(FaqTopicType.BenefitsAndCoverage),
        );
      case 'Claims':
        return setTopics(SupportFaqTopicDetails.get(FaqTopicType.Claims));

      case 'ID Cards':
        return setTopics(SupportFaqTopicDetails.get(FaqTopicType.IdCards));

      case 'My Plan Information':
        return setTopics(
          SupportFaqTopicDetails.get(FaqTopicType.MyPlanInformation),
        );

      case 'Pharmacy':
        return setTopics(SupportFaqTopicDetails.get(FaqTopicType.Pharmacy));

      case 'Prior Authorization':
        return setTopics(
          SupportFaqTopicDetails.get(FaqTopicType.PriorAuthorization),
        );

      case 'Sharing, Permissions & Security':
        return setTopics(
          SupportFaqTopicDetails.get(FaqTopicType.SharingPermisionsSecurity),
        );
    }
  };

  return (
    <Column className="app-content app-base-font-color">
      <FaqHeaderCard
        faqHeaderDetails={
          topics?.faqTopicHeaderDetails ?? ({} as FaqHeaderCardDetails)
        }
      />
      <Spacer size={16} />
      <section className="flex flex-row items-start app-body">
        <Column className="flex-grow page-section-63_33 items-stretch">
          <FaqCard
            services={topics?.faqTopCardDetails ?? []}
            topicType={topics?.topicType}
          />
          <FaqCard
            services={topics?.faqBottomCardDetails}
            topicType={topics?.topicType}
          />
        </Column>
        <Column className=" flex-grow page-section-36_67 items-stretch ">
          <OtherFaqTopics
            faqTopics={OtherFaqTopicDetails}
            goToFaqPage={activeFaqTopic}
          />
        </Column>
      </section>
    </Column>
  );
};
