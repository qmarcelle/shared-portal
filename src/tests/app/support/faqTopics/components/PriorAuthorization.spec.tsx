/* eslint-disable quotes */
import { FaqTopics } from '@/app/support/faqTopics/components/FaqTopics';
import { fireEvent, render, screen } from '@testing-library/react';

const renderUI = () => {
  return render(<FaqTopics />);
};

describe('Prior Authorization FAQ', () => {
  it('should render the UI correctly', async () => {
    const component = renderUI();
    const priorAuth = screen.getByRole('button', {
      name: 'Prior Authorization',
    });

    fireEvent.click(priorAuth);
    screen.getByText('Learn more about prior authorizations and statuses.');

    screen.getByRole('heading', {
      name: 'Understanding Prior Authorization',
    });
    screen.getByText('What is a prior authorization?');
    fireEvent.click(screen.getByText('What is a prior authorization?'));
    screen.getByText(
      "Prior authorization is a process that ensures any prescribed treatments or drugs are appropriate and safe. Prior authorizations require your health provider to get approval from your health insurance plan. It's also known as preauthorization or pre-certification.",
    );

    screen.getByText('How do I get a prior authorization?');
    fireEvent.click(screen.getByText('How do I get a prior authorization?'));
    screen.getByText(
      'If your doctor prescribes a treatment that needs prior authorization, they’ll ask us for one when they write your treatment plan. We’ll review the case, make sure the treatment is appropriate and safe, and make a decision. If it’s approved, we’ll send you a letter to let you know.',
    );

    screen.getByRole('heading', {
      name: 'Help with Prior Authorization Status',
    });
    screen.getByText('What does the status of my prior authorizations mean?');
    fireEvent.click(
      screen.getByText('What does the status of my prior authorizations mean?'),
    );
    screen.getByText(
      'Prior authorizations can have one of three statuses: approved, partially approved or denied. Approved means that your health plan agrees with your doctor that the medical service or prescription drug is necessary and safe. Partially approved means that your health plan has concerns about the safety or need of part of your request. Denied means that your health plan does not agree with your doctor that a medical service or prescription drug is necessary or safe.',
    );

    screen.getByText('My prior authorization was approved. What do I do next?');
    fireEvent.click(
      screen.getByText(
        'My prior authorization was approved. What do I do next?',
      ),
    );
    screen.getByText(
      'If your request was approved, move forward with the medical service or drug as your doctor instructed.',
    );

    screen.getByText(
      'What can I do if my prior authorization is partially approved or denied?',
    );

    fireEvent.click(
      screen.getByText(
        'What can I do if my prior authorization is partially approved or denied?',
      ),
    );

    screen.getByText(
      "I'm expecting a prior authorization. Where are all my prior authorizations?",
    );

    fireEvent.click(
      screen.getByText(
        "I'm expecting a prior authorization. Where are all my prior authorizations?",
      ),
    );

    screen.getByRole('heading', {
      name: 'Other FAQ Topics',
    });
    screen.getByText('Benefits & Coverage');
    screen.getByText('Claims');
    screen.getByText('ID Cards');
    screen.getByText('My Plan Information');
    screen.getByText('Pharmacy');
    screen.getByText('Prior Authorization');
    screen.getByText('Security');

    expect(component.container).toMatchSnapshot();
  });
});
