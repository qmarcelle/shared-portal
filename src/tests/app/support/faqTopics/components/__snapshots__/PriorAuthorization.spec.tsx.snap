// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prior Authorization FAQ should render the UI correctly 1`] = `
<div>
  <div
    class="flex flex-col app-content app-base-font-color"
  >
    <div
      class="flex flex-col md:m-2 m-4"
    >
      <div
        class="flex flex-row justify-between mt-2 md:mt-16 lg:mt-0"
      >
        <div
          class="flex flex-col max-w-2xl"
        >
          <span
            style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
          />
          <h1
            class="title-1"
          >
            Prior Authorization FAQ
          </h1>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
          />
          <p
            class="body-1 body-1"
          >
            Learn more about prior authorizations and statuses.
          </p>
        </div>
      </div>
    </div>
    <section
      class="flex flex-row items-start app-body"
    >
      <div
        class="flex flex-col flex-grow page-section-63_33 items-stretch"
      >
        <div
          class="card-main large-section flex flex-row items-start app-body "
        >
          <div
            class="flex flex-col w-[100%]"
          >
             
            <div
              class="flex flex-col"
            >
              <h2
                class="title-2"
              >
                Understanding Prior Authorization
              </h2>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
              />
              <div
                class="flex flex-col items-stretch"
              >
                <div
                  class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-2 "
                >
                  <div
                    class="flex flex-row m-1 accordion-head"
                  >
                    <p
                      class="mr-2 flex-grow"
                    >
                      What is a prior authorization?
                    </p>
                    <img
                      alt="Up Chevron"
                      class="pl-2 w-6"
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                  </div>
                  <div
                    class="m-1"
                  >
                    <div
                      class="flex flex-col flex flex-col"
                    >
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                      />
                      <div
                        class="flex flex-row"
                      >
                        <span
                          class="rich-text body-1 undefined"
                        >
                          <span>
                            Prior authorization is a process that ensures any prescribed treatments or drugs are appropriate and safe. Prior authorizations require your health provider to get approval from your health insurance plan. It's also known as preauthorization or pre-certification.
                          </span>
                        </span>
                      </div>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
                      />
                      <div
                        class="flex flex-row"
                      >
                        <ul
                          class="ml-4"
                        />
                      </div>
                      <div
                        class="flex flex-row"
                      >
                        <span
                          class="rich-text body-1 undefined"
                        >
                          <span />
                        </span>
                      </div>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class=""
                  style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                />
              </div>
            </div>
            <div
              class="flex flex-col"
            >
              <h2
                class="title-2"
              />
              <div
                class="flex flex-col items-stretch"
              >
                <div
                  class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-2 "
                >
                  <div
                    class="flex flex-row m-1 accordion-head"
                  >
                    <p
                      class="mr-2 flex-grow"
                    >
                      How do I get a prior authorization?
                    </p>
                    <img
                      alt="Up Chevron"
                      class="pl-2 w-6"
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                  </div>
                  <div
                    class="m-1"
                  >
                    <div
                      class="flex flex-col flex flex-col"
                    >
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                      />
                      <div
                        class="flex flex-row"
                      >
                        <span
                          class="rich-text body-1 undefined"
                        >
                          <span>
                            If your doctor prescribes a treatment that needs prior authorization, they’ll ask us for one when they write your treatment plan. We’ll review the case, make sure the treatment is appropriate and safe, and make a decision. If it’s approved, we’ll send you a letter to let you know.
                          </span>
                        </span>
                      </div>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
                      />
                      <div
                        class="flex flex-row"
                      >
                        <ul
                          class="ml-4"
                        />
                      </div>
                      <div
                        class="flex flex-row"
                      >
                        <span
                          class="rich-text body-1 undefined"
                        >
                          <span />
                        </span>
                      </div>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="card-main large-section flex flex-row items-start app-body "
        >
          <div
            class="flex flex-col w-[100%]"
          >
             
            <div
              class="flex flex-col"
            >
              <h2
                class="title-2"
              >
                Help with Prior Authorization Status
              </h2>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
              />
              <div
                class="flex flex-col items-stretch"
              >
                <div
                  class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-2 "
                >
                  <div
                    class="flex flex-row m-1 accordion-head"
                  >
                    <p
                      class="mr-2 flex-grow"
                    >
                      What does the status of my prior authorizations mean?
                    </p>
                    <img
                      alt="Up Chevron"
                      class="pl-2 w-6"
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                  </div>
                  <div
                    class="m-1"
                  >
                    <div
                      class="flex flex-col flex flex-col"
                    >
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                      />
                      <div
                        class="flex flex-row"
                      >
                        <span
                          class="rich-text body-1 undefined"
                        >
                          <span>
                            Prior authorizations can have one of three statuses: approved, partially approved or denied. Approved means that your health plan agrees with your doctor that the medical service or prescription drug is necessary and safe. Partially approved means that your health plan has concerns about the safety or need of part of your request. Denied means that your health plan does not agree with your doctor that a medical service or prescription drug is necessary or safe.
                          </span>
                        </span>
                      </div>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
                      />
                      <div
                        class="flex flex-row"
                      >
                        <ul
                          class="ml-4"
                        />
                      </div>
                      <div
                        class="flex flex-row"
                      >
                        <span
                          class="rich-text body-1 undefined"
                        >
                          <span />
                        </span>
                      </div>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class=""
                  style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                />
              </div>
            </div>
            <div
              class="flex flex-col"
            >
              <h2
                class="title-2"
              />
              <div
                class="flex flex-col items-stretch"
              >
                <div
                  class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-2 "
                >
                  <div
                    class="flex flex-row m-1 accordion-head"
                  >
                    <p
                      class="mr-2 flex-grow"
                    >
                      My prior authorization was approved. What do I do next?
                    </p>
                    <img
                      alt="Up Chevron"
                      class="pl-2 w-6"
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                  </div>
                  <div
                    class="m-1"
                  >
                    <div
                      class="flex flex-col flex flex-col"
                    >
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                      />
                      <div
                        class="flex flex-row"
                      >
                        <span
                          class="rich-text body-1 undefined"
                        >
                          <span>
                            If your request was approved, move forward with the medical service or drug as your doctor instructed.
                          </span>
                        </span>
                      </div>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
                      />
                      <div
                        class="flex flex-row"
                      >
                        <ul
                          class="ml-4"
                        />
                      </div>
                      <div
                        class="flex flex-row"
                      >
                        <span
                          class="rich-text body-1 undefined"
                        >
                          <span />
                        </span>
                      </div>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class=""
                  style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                />
              </div>
            </div>
            <div
              class="flex flex-col"
            >
              <h2
                class="title-2"
              />
              <div
                class="flex flex-col items-stretch"
              >
                <div
                  class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-2 "
                >
                  <div
                    class="flex flex-row m-1 accordion-head"
                  >
                    <p
                      class="mr-2 flex-grow"
                    >
                      What can I do if my prior authorization is partially approved or denied?
                    </p>
                    <img
                      alt="Up Chevron"
                      class="pl-2 w-6"
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                  </div>
                  <div
                    class="m-1"
                  >
                    <div
                      class="flex flex-col flex flex-col"
                    >
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                      />
                      <div
                        class="flex flex-row"
                      >
                        <span
                          class="rich-text body-1 undefined"
                        >
                          <span>
                            <span
                              class="rich-text body-1 undefined"
                            >
                              <span>
                                If a medical service or prescription drug prior authorization was partially approved or denied, we’ll send you a letter explaining why with details on how to file an appeal. For more information, please
                              </span>
                              <span>
                                <a
                                  class="link font-bold"
                                >
                                   contact us.
                                </a>
                              </span>
                            </span>
                          </span>
                        </span>
                      </div>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
                      />
                      <div
                        class="flex flex-row"
                      >
                        <ul
                          class="ml-4"
                        />
                      </div>
                      <div
                        class="flex flex-row"
                      >
                        <span
                          class="rich-text body-1 undefined"
                        >
                          <span />
                        </span>
                      </div>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class=""
                  style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                />
              </div>
            </div>
            <div
              class="flex flex-col"
            >
              <h2
                class="title-2"
              />
              <div
                class="flex flex-col items-stretch"
              >
                <div
                  class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-2 "
                >
                  <div
                    class="flex flex-row m-1 accordion-head"
                  >
                    <p
                      class="mr-2 flex-grow"
                    >
                      I'm expecting a prior authorization. Where are all my prior authorizations?
                    </p>
                    <img
                      alt="Up Chevron"
                      class="pl-2 w-6"
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                  </div>
                  <div
                    class="m-1"
                  >
                    <div
                      class="flex flex-col flex flex-col"
                    >
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                      />
                      <div
                        class="flex flex-row"
                      >
                        <span
                          class="rich-text body-1 undefined"
                        >
                          <span>
                            <span
                              class="rich-text body-1 undefined"
                            >
                              <span>
                                We’ll show your medical prior authorizations on the
                              </span>
                              <span>
                                <a
                                  class="link font-bold"
                                  href="/priorAuthorization"
                                >
                                   
                                  prior authorization page.
                                </a>
                              </span>
                              <span>
                                 
                                For any prescription drug-related prior authorizations, you may need to log in to your Caremark account. If your pharmacy benefits aren’t provided by CVS Caremark, you can
                              </span>
                              <span>
                                <a
                                  class="link font-bold"
                                >
                                   contact us
                                </a>
                              </span>
                              <span>
                                , and we’ll help you find your prescription prior authorization status.
                              </span>
                            </span>
                          </span>
                        </span>
                      </div>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
                      />
                      <div
                        class="flex flex-row"
                      >
                        <ul
                          class="ml-4"
                        />
                      </div>
                      <div
                        class="flex flex-row"
                      >
                        <span
                          class="rich-text body-1 undefined"
                        >
                          <span />
                        </span>
                      </div>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col  flex-grow page-section-36_67 items-stretch"
      >
        <div
          class="card-main mt-4"
        >
          <div
            class="flex flex-col ml-6"
          >
            <span
              style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
            />
            <h2
              class="title-2 !font-light !text-[26px]/[40px] ml-4"
            >
              Other FAQ Topics
            </h2>
            <span
              style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
            />
            <div
              class="flex flex-row"
            >
              <ul
                class="ml-6 !font-thin"
              >
                <div
                  class="flex flex-col !font-thin"
                >
                  <li>
                    <div
                      class="flex flex-row"
                    >
                      <a
                        aria-label="Benefits & Coverage"
                        style="display: block;"
                        tabindex="-1"
                      >
                        <button
                          class="flex flex-row link-container p-0 "
                          style="max-width: max-content; height: auto; display: block;"
                          tabindex="-1"
                          type="button"
                        >
                          <p
                            class="link !no-underline manage-faq"
                            style="display: block;"
                          >
                            Benefits & Coverage
                          </p>
                        </button>
                      </a>
                    </div>
                  </li>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                  />
                </div>
                <div
                  class="flex flex-col !font-thin"
                >
                  <li>
                    <div
                      class="flex flex-row"
                    >
                      <a
                        aria-label="Claims"
                        style="display: block;"
                        tabindex="-1"
                      >
                        <button
                          class="flex flex-row link-container p-0 "
                          style="max-width: max-content; height: auto; display: block;"
                          tabindex="-1"
                          type="button"
                        >
                          <p
                            class="link !no-underline manage-faq"
                            style="display: block;"
                          >
                            Claims
                          </p>
                        </button>
                      </a>
                    </div>
                  </li>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                  />
                </div>
                <div
                  class="flex flex-col !font-thin"
                >
                  <li>
                    <div
                      class="flex flex-row"
                    >
                      <a
                        aria-label="ID Cards"
                        style="display: block;"
                        tabindex="-1"
                      >
                        <button
                          class="flex flex-row link-container p-0 "
                          style="max-width: max-content; height: auto; display: block;"
                          tabindex="-1"
                          type="button"
                        >
                          <p
                            class="link !no-underline manage-faq"
                            style="display: block;"
                          >
                            ID Cards
                          </p>
                        </button>
                      </a>
                    </div>
                  </li>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                  />
                </div>
                <div
                  class="flex flex-col !font-thin"
                >
                  <li>
                    <div
                      class="flex flex-row"
                    >
                      <a
                        aria-label="My Plan Information"
                        style="display: block;"
                        tabindex="-1"
                      >
                        <button
                          class="flex flex-row link-container p-0 "
                          style="max-width: max-content; height: auto; display: block;"
                          tabindex="-1"
                          type="button"
                        >
                          <p
                            class="link !no-underline manage-faq"
                            style="display: block;"
                          >
                            My Plan Information
                          </p>
                        </button>
                      </a>
                    </div>
                  </li>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                  />
                </div>
                <div
                  class="flex flex-col !font-thin"
                >
                  <li>
                    <div
                      class="flex flex-row"
                    >
                      <a
                        aria-label="Pharmacy"
                        style="display: block;"
                        tabindex="-1"
                      >
                        <button
                          class="flex flex-row link-container p-0 "
                          style="max-width: max-content; height: auto; display: block;"
                          tabindex="-1"
                          type="button"
                        >
                          <p
                            class="link !no-underline manage-faq"
                            style="display: block;"
                          >
                            Pharmacy
                          </p>
                        </button>
                      </a>
                    </div>
                  </li>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                  />
                </div>
                <div
                  class="flex flex-col !font-thin"
                >
                  <li>
                    <div
                      class="flex flex-row"
                    >
                      <a
                        aria-label="Prior Authorization"
                        style="display: block;"
                        tabindex="-1"
                      >
                        <button
                          class="flex flex-row link-container border-l-4  focus:outline-none border-primary font-bold p-0 pl-1"
                          style="max-width: max-content; height: auto; display: block;"
                          tabindex="-1"
                          type="button"
                        >
                          <p
                            class="link !no-underline manage-faq"
                            style="display: block;"
                          >
                            Prior Authorization
                          </p>
                        </button>
                      </a>
                    </div>
                  </li>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                  />
                </div>
                <div
                  class="flex flex-col !font-thin"
                >
                  <li>
                    <div
                      class="flex flex-row"
                    >
                      <a
                        aria-label="Security"
                        style="display: block;"
                        tabindex="-1"
                      >
                        <button
                          class="flex flex-row link-container p-0 "
                          style="max-width: max-content; height: auto; display: block;"
                          tabindex="-1"
                          type="button"
                        >
                          <p
                            class="link !no-underline manage-faq"
                            style="display: block;"
                          >
                            Security
                          </p>
                        </button>
                      </a>
                    </div>
                  </li>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                  />
                </div>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
